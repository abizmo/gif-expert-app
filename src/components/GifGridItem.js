import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({ img, title, username }) => {
  return (
    <div className="card animate__animated animate__fadeInBottomLeft">
      <img src={img} alt={title} />
      <div className="info">
        <h5>{title}</h5>
        <p>{username}</p>
      </div>
    </div>
  )
}

GifGridItem.propTypes = {
  img: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
}

export default GifGridItem
