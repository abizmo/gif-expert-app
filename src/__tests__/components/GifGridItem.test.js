import { shallow } from "enzyme"
import GifGridItem from "../../components/GifGridItem"

describe('GifGridItem Tests', () => {
  const img = 'https://media1.giphy.com/media/xUA7aLrdFvkS3BeaEE/giphy.gif?cid=e23a9f3bdbm7ugc2t3ryb9qt5b2kazo22d1hf6mq4oxudl8k&rid=giphy.gif&ct=g'
  const title = 'Little turds'
  const username = 'teamcoco'
  const wrapper = shallow(<GifGridItem img={img} title={title} username={username} />)
  test('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('should render the title inside a h5', () => {
    const h5Text = wrapper.find('h5').text()
    expect(h5Text).toBe(title);
  })

  test('should render the username inside a p', () => {
    const pText = wrapper.find('p').text()
    expect(pText).toBe(username);
  })

  test('should render the image with src=img and alt=title', () => {
    const { src, alt } = wrapper.find('img').props()
    expect(src).toBe(img);
    expect(alt).toBe(title);
  })

  test('should have animate__fadeInBottomLeft class', () => {
    const className = 'animate__fadeInBottomLeft'
    const classDiv = wrapper.find('div').at(0).prop('className');
    expect(classDiv).toContain(className)
  })
  
  
})
