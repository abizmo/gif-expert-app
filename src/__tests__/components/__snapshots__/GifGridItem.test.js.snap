// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Tests should render the component 1`] = `
<div
  className="card animate__animated animate__fadeInBottomLeft"
>
  <img
    alt="Little turds"
    src="https://media1.giphy.com/media/xUA7aLrdFvkS3BeaEE/giphy.gif?cid=e23a9f3bdbm7ugc2t3ryb9qt5b2kazo22d1hf6mq4oxudl8k&rid=giphy.gif&ct=g"
  />
  <div
    className="info"
  >
    <h5>
      Little turds
    </h5>
    <p>
      teamcoco
    </p>
  </div>
</div>
`;
